core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -615,6 +672,93 @@ static void set_user_va_idx(struct mmu_partition *prtn)
 	assert(user_va_idx != -1);
 }
 
+/*
+ * Setup an entry inside a core level 1 page table for TAs memory mapping
+ *
+ * If base table level is 1 - user_va_idx is already the index,
+ *                            so nothing to do.
+ * If base table level is 0 - we might need to allocate entry 0 of base table,
+ *                            as TAs page table is an entry inside a level 1
+ *                            page table.
+ */
+static void core_init_mmu_prtn_ta_core(struct mmu_partition *prtn
+				       __maybe_unused,
+				       unsigned int base_idx __maybe_unused,
+				       unsigned int core __maybe_unused)
+{
+#if (CORE_MMU_BASE_TABLE_LEVEL == 0)
+	uint64_t *tbl = NULL;
+	uintptr_t idx = 0;
+
+	assert(user_va_idx != -1);
+	COMPILE_TIME_ASSERT(MAX_XLAT_TABLES <
+			    (1 << (8 * sizeof(prtn->user_l1_table_idx[0][0]))));
+
+	tbl = prtn->base_tables[base_idx][core];
+
+	/*
+	 * If base level is 0, then user_va_idx refers to
+	 * level 1 page table that's in base level 0 entry 0.
+	 */
+	if (CORE_MMU_BASE_TABLE_LEVEL == 0) {
+		struct core_mmu_table_info tbl_info = { };
+		bool ret = false;
+
+		core_mmu_set_info_table(&tbl_info, 0, 0, tbl);
+#ifdef CFG_VIRTUALIZATION
+		tbl_info.prtn = prtn;
+#endif
+
+		/*
+		 * If this isn't the core that created the initial tables
+		 * mappings, then the level 1 table must be copied,
+		 * as it will hold pointer to the user mapping table
+		 * that changes per core.
+		 */
+		if (core != get_core_pos()) {
+			ret = core_mmu_entry_copy(&tbl_info, 0);
+			assert(ret);
```

Comment:
If xlat tables pool is exhausted, `core_mmu_entry_copy()` returns `false`. We can `assert()` here, must use an non debug test: `if (!core_mmu_entry_copy(&tbl_info, 0)) { panic(); }`

Note I think it would be better if `core_mmu_entry_copy()` could return a `TEE_Result` value instead of a boolean but that's not a strong opinion (as i.e. `core_mmu_entry_to_finer_grained()` already returns a boolean).

Reply:
replaced assert to panic. didn't change the return value.